var/spack/repos/builtin/packages/python/package.py

```diff
@@ -202,8 +202,9 @@ def site_packages_dir(self):
     def setup_dependent_environment(self, spack_env, run_env, extension_spec):
         """Set PYTHONPATH to include site-packages dir for the
         extension and any other python extensions it depends on."""
-        pythonhome = self.prefix
-        spack_env.set('PYTHONHOME', pythonhome)
+        python = Executable(os.path.join(self.prefix, 'python'))
```

Comment:
In `setup_dependent_package()`, we use syntax like:
```python
'python{0}'.format('3' if self.spec.satisfies('@3') else '')
```
This could easily be used here too. I wonder if we can make it a more generic property.